### ISSUE TITLE
Add ability to specify a role on a view container for accessibility

### ISSUE THREAD

**Issue by vritant24** [microsoft/vscode#257686]
## Problem
VSCode for Education today uses the view container to host a tree view that is used for navigation by the user through the course. 
<img width="1124" height="1065" alt="Image" src="https://github.com/user-attachments/assets/8bdfb44f-1de8-4666-8d88-68814ab1a085" />

**According to accessibility guidelines, we need to provide a `nav` role to the container containing the TreeView**

As a result, the lack of navigation landmarks hinders users who rely on screen readers from understanding the page structure and efficiently navigating the interface.

**Steps to Reproduce**:
  1. Visit [https://vscodeedu.com/](https://vscodeedu.com/).
  2. Sign in with a Microsoft account.
  3. Navigate to the "Courses" tab, select a course, and select "Start."
  4. Enable the Landmarks toggle in Accessibility Insights for Web.
  5. Observe that a Navigation landmark is missing from the TreeView area.

## Request
Have the ability to specify a role for a view container support [landmarks](https://www.w3.org/WAI/ARIA/apg/practices/landmark-regions/) (like `nav`, `search`, etc)
